### PR TITLE
Add feed mapper

### PR DESCRIPTION
**Pull request by snapshotpl**

### CODE REVIEW

File: module/Application/src/Application/Controller/IndexController.php [hunk 1]
```diff
@@ -85,4 +73,14 @@ public function feedAction()
 
         return $feedmodel;
     }
+
+    /**
+     * @param int $page
+     * @param string $query
+     * @return \Zend\Paginator\Paginator
+     */
+    protected function getModules($page = 1, $query = null)
```
Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: i think this method is not necessary and the name don't indicates there actually purpose - there is no need to wrap it into a sep. method

**Comment by ins0** [comment]
@snapshotpl can u please provide tests for this feature?

**Comment by snapshotpl** [comment]
@ins0 Test provided, but Travis looks dead ;)

File: module/Application/src/Application/Controller/IndexController.php [hunk 2]
```diff
@@ -52,32 +56,17 @@ public function feedAction()
         // Prepare the feed
         $feed = new Feed();
         $feed->setTitle('ZF2 Modules');
-        $feed->setDescription('Recently added modules.');
+        $feed->setDescription('Recently added ZF2 modules');
         $feed->setFeedLink('http://modules.zendframework.com/feed', 'atom');
         $feed->setLink('http://modules.zendframework.com');
```
Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: could you please update the url too? ``zfmodules.com``

File: module/Application/src/Application/Controller/IndexController.php [hunk 3]
```diff
@@ -52,32 +56,17 @@ public function feedAction()
         // Prepare the feed
         $feed = new Feed();
         $feed->setTitle('ZF2 Modules');
-        $feed->setDescription('Recently added modules.');
+        $feed->setDescription('Recently added ZF2 modules');
```
Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: could you please update the url too? ``zfmodules.com``

File: module/Application/src/Application/Controller/IndexController.php (hunk 2 above)
Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: could you please update the url too? ``zfmodules.com``

Comment (snapshotpl) [inline module/Application/src/Application/Controller/IndexController.php]: I think we should create a new PR with this update in whole project

**Comment by ins0** [comment]
thanks @snapshotpl for providing tests :+1: - if the url from the comment above gets updated this looks good from my point of view :smiley:

File: module/Application/src/Application/Controller/IndexController.php (hunk 2 above)
Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: yeah sounds like a better plan :100:

Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: actually this is the only file that contains the url and needs an update ^^

Comment (snapshotpl) [inline module/Application/src/Application/Controller/IndexController.php]: There is also link in README

Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: missed that thanks :+1:

Comment (snapshotpl) [inline module/Application/src/Application/Controller/IndexController.php]: Or, we should use here serverUrl helper to be more elastic, what are you think?

Comment (gianarb) [inline module/Application/src/Application/Controller/IndexController.php]: Maybe `serverUrl` is a best solution

Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: Isn't this a View Helper?

Comment (gianarb) [inline module/Application/src/Application/Controller/IndexController.php]: mmm you are in Controller..
You can use
```php
$url = $this->url()->fromRoute('home', [], ['force_canonical', true]);
```
@ins0 prints the url into the application in my opinion is not a good idea.. But I don't remember if this solution works

Comment (ins0) [inline module/Application/src/Application/Controller/IndexController.php]: @gianarb :+1:

Comment (snapshotpl) [inline module/Application/src/Application/Controller/IndexController.php]: Change in progress...

**Comment by snapshotpl** [comment]
@ins0 @gianarb Done. I extended also feed entry data

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    /**
+     * @var Url
+     */
+    protected $url;
+
+    /**
+     * @param Feed $feed
+     */
+    public function __construct(Feed $feed, Url $url)
```
Comment (ins0) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: i would suggest to name this more explicit and not simple ``$url``

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        $moduleName = $module->getName();
+        $urlParams = ['vendor' => $module->getOwner(), 'module' => $moduleName];
+
+        $entry = $this->feed->createEntry();
+        $entry->setTitle($moduleName);
+        $entry->setDescription($moduleDescription);
+        $entry->setLink($this->url->fromRoute('view-module', $urlParams, ['force_canonical' => true]));
+        $entry->addAuthor(['name' => $module->getOwner()]);
+        $entry->setDateCreated($module->getCreatedAtDateTime());
+        $entry->setId(implode('/', $urlParams));
```
Comment (ins0) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: this looks confused as you don't see in the first place what id is being set here

**Comment by snapshotpl** [comment]
@ins0 what about now? :)

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    public function addModule(ModuleEntity $module)
+    {
+        if (empty($module->getDescription())) {
+            $moduleDescription = 'No description available';
+        } else {
+            $moduleDescription = $module->getDescription();
+        }
+        $moduleName = $module->getName();
+        $urlParams = ['vendor' => $module->getOwner(), 'module' => $moduleName];
+        $id = implode('/', $urlParams);
```
Comment (ins0) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: this is equal odd. how about ``sprintf('%s/%s',  $module->getOwner(), $moduleName)`` or we extend the module entity to get a full repository name - but not sure about that. /cc @Ocramius @localheinz

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+    /**
+     * @param ModuleEntity $module
+     * @return Entry
+     */
+    public function addModule(ModuleEntity $module)
+    {
+        if (empty($module->getDescription())) {
+            $moduleDescription = 'No description available';
+        } else {
```
Comment (ins0) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: you can skip this ``else`` and declare the first set before the ``if(empty(...``

Comment (snapshotpl) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: Of course I can, but are we really need two assigns to `$moduleDescription`?

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php (hunk 6 above)
Comment (snapshotpl) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: I like idea with additional method i module entity. We can reuse it in views.

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php (hunk 7 above)
Comment (ins0) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: ocramius suggested this in my pr too, it is cleaner to read - but i'm fine if this stays :smiley:

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php (hunk 6 above)
Comment (ins0) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: yeah i ike this idea as well, there should be no reason to follow this way :+1:

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php (hunk 7 above)
Comment (Ocramius) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: Yes, two assignments is good

File: module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php (hunk 6 above)
Comment (Ocramius) [inline module/ZfModule/src/ZfModule/Mapper/ModuleToFeed.php]: If the module pseudo-identifier is this one, then yes, this should be part of the `Module` model API

**Comment by snapshotpl** [comment]
My PR needs review :-)

**Comment by snapshotpl** [comment]
@ins0 who said that Module entity can come from db?

**Comment by Ocramius** [comment]
Merged, thanks @snapshotpl!

**Pull request merged**